engine/apps/slack/tasks.py

```diff
@@ -32,39 +32,102 @@
+    - Upon successful completion, clears the `active_update_task_id` to allow future updates.
+
+    Args:
+        slack_message_pk (int): The primary key of the `SlackMessage` instance to update.
+    """
+    from apps.slack.models import SlackMessage
+
+    current_task_id = update_alert_group_slack_message.request.id
+
+    logger.info(
+        f"update_alert_group_slack_message for slack message {slack_message_pk} started with task_id {current_task_id}"
+    )
+
+    try:
+        slack_message = SlackMessage.objects.get(pk=slack_message_pk)
+    except SlackMessage.DoesNotExist:
+        logger.warning(f"SlackMessage {slack_message_pk} doesn't exist")
+        return
+
+    if current_task_id != slack_message.active_update_task_id:
+        logger.warning(
+            f"update_alert_group_slack_message skipped, because current_task_id ({current_task_id}) "
+            f"does not equal to active_update_task_id ({slack_message.active_update_task_id}) "
         )
-        cache.set(cache_key, update_task_id, timeout=CACHE_UPDATE_INCIDENT_SLACK_MESSAGE_LIFETIME)
+        return
 
-        return (
-            f"update_incident_slack_message rescheduled because of current task_id ({current_task_id})"
-            f" for alert_group {alert_group_pk} doesn't exist in cache"
+    alert_group = slack_message.alert_group
+    if not alert_group:
+        logger.warning(
+            f"skipping update_alert_group_slack_message as SlackMessage {slack_message_pk} "
+            "doesn't have an alert group associated with it"
         )
-    if not current_task_id == cached_task_id:
-        return (
-            f"update_incident_slack_message skipped, because of current task_id ({current_task_id})"
-            f" doesn't equal to cached task_id ({cached_task_id}) for alert_group {alert_group_pk}"
+        return
+
+    slack_client = SlackClient(slack_message.slack_team_identity)
+
+    alert_group_pk = alert_group.pk
+    alert_receive_channel = alert_group.channel
+
+    try:
+        slack_client.chat_update(
+            # TODO: once _channel_id has been fully migrated to channel, remove _channel_id
+            # see https://raintank-corp.slack.com/archives/C06K1MQ07GS/p173255546
+            # channel=slack_message.channel.slack_id,
+            channel=slack_message._channel_id,
+            ts=slack_message.slack_id,
+            attachments=alert_group.render_slack_attachments(),
+            blocks=alert_group.render_slack_blocks(),
         )
 
+        logger.info(f"Message has been updated for alert_group {alert_group_pk}")
+    except SlackAPIRatelimitError as e:
+        if not alert_receive_channel.is_maintenace_integration:
+            if not alert_receive_channel.is_rate_limited_in_slack:
+                alert_receive_channel.start_send_rate_limit_message_task("Updating", e.retry_after)
+                logger.info(f"Message has not been updated for alert_group {alert_group_pk} due to slack rate limit.")
```

Comment:
I think this should come before the `slack_client.chat_update` call? (at least it's the case for the old version)
I _think_ the idea behind that is to make sure we don't hammer Slack with API calls when we already know we have been ratelimited.

Reply:
good point 👍 

addressed in https://github.com/grafana/oncall/pull/5287/commits/82db6a5b8692ce5b514f277446f000bf200a9976